rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/NodeSourceConfigurationParser.java

```diff
@@ -27,60 +27,73 @@
+            throw new IllegalArgumentException("The imported node source is not a valid JSON file", e);
         }
 
         try {
             String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
             boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
 
-            JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
-            String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName")
-                                                                                .isString()
-                                                                                .stringValue();
-            PluginDescriptor infrastructurePluginDescriptor = getPluginDescriptor(infrastructurePluginDescriptorJson,
-                                                                                  infrastructurePluginName);
-
-            JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
-            String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
-            PluginDescriptor policyPluginDescriptor = getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+            PluginDescriptor infrastructurePluginDescriptor = getInfrastructurePluginDescriptor(jsonObject);
+            PluginDescriptor policyPluginDescriptor = getPolicyPluginDescriptor(jsonObject);
 
             return new NodeSourceConfiguration(nodeSourceName,
                                                nodesRecoverable,
                                                infrastructurePluginDescriptor,
                                                policyPluginDescriptor);
         } catch (RuntimeException e) {
-            throw new ImportException("The imported node source has incorrect parameters.", e);
+            throw new IllegalArgumentException("The imported node source has incorrect parameters.", e);
         }
     }
 
+    private PluginDescriptor getInfrastructurePluginDescriptor(JSONObject jsonObject) {
+        JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+        String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName").isString().stringValue();
+        return getPluginDescriptor(infrastructurePluginDescriptorJson, infrastructurePluginName);
+    }
+
+    private PluginDescriptor getPolicyPluginDescriptor(JSONObject jsonObject) {
+        JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
+        String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
+        return getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+    }
+
     public HashMap<String, PluginDescriptor> parsePluginDescriptors(String json) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)